contracts/core/libraries/QVHelper.sol

```diff
@@ -0,0 +1,100 @@
+
+    /// @notice Votes for recipients
+    /// @param _state The voting state
+    /// @param _recipients The recipients to vote
+    /// @param _voiceCredits The amounts of voice credits to for each recipient
+    /// @dev The number of recipients and voiceCredits should be equal and the same index should correspond to the same recipient and amount
+    function _voteWithVoiceCredits(
+        VotingState storage _state,
+        address[] memory _recipients,
+        uint256[] memory _voiceCredits
+    ) internal {
+        /// Check if the number of recipients and amounts are equal
+        if (_recipients.length != _voiceCredits.length) revert QVHelper_LengthMissmatch();
+
+        for (uint256 i = 0; i < _recipients.length; i++) {
+            /// Add the voice credits to the recipient
+            _state.recipientVoiceCredits[_recipients[i]] += _voiceCredits[i];
+            uint256 _votes = FixedPointMathLib.sqrt(_voiceCredits[i] * 1e18);
+            /// Add the votes to the recipient
+            _state.recipientVotes[_recipients[i]] += _votes;
+            /// Add the total voice credits
+            _state.totalVoiceCredits += _voiceCredits[i];
+            /// Add the total votes
+            _state.totalVotes += _votes;
+        }
+    }
+
+    /// @notice Votes for recipients
+    /// @param _state The voting state
+    /// @param _recipients The recipients to vote
+    /// @param _votes The amounts of votes to for each recipient
+    /// @dev The number of recipients and votes should be equal and the same index should correspond to the same recipient and amount
+    function _vote(VotingState storage _state, address[] memory _recipients, uint256[] memory _votes) internal {
+        /// Check if the number of recipients and amounts are equal
+        if (_recipients.length != _votes.length) revert QVHelper_LengthMissmatch();
+
+        for (uint256 i = 0; i < _recipients.length; i++) {
+            /// Add the votes to the recipient
+            _state.recipientVotes[_recipients[i]] += _votes[i];
+            /// Add the total votes
+            _state.totalVotes += _votes[i];
+            /// voiceCredits = votes^2
+            uint256 _voiceCredits = _votes[i] * _votes[i];
+            /// Add the voice credits to the recipient
+            _state.recipientVoiceCredits[_recipients[i]] += _voiceCredits;
+            /// Add total voice credits
+            _state.totalVoiceCredits += _voiceCredits;
+        }
+    }
+
+    /// @notice Calculate the payout for each recipient
+    /// @param _state The state of the Quadratic Voting
+    /// @param _recipients The recipients
+    /// @param _poolAmount The amount of the pool
+    /// @return __recipients The recipients
+    /// @return _payouts The payouts for each recipient
+    function _getPayout(VotingState storage _state, address[] memory _recipients, uint256 _poolAmount)
+        internal
+        view
+        returns (address[] memory __recipients, uint256[] memory _payouts)
```

Comment:
What's the reason for returning `_recipients`? Can we remove it or at least optimize it?